frontend/src/scenes/data-warehouse/editor/editorSidebarLogic.tsx

```diff
@@ -62,139 +57,273 @@ export const editorSidebarLogic = kea<editorSidebarLogicType>([
-                                      actions.selectSchema(table)
-                                  },
-                                  menuItems: [
-                                      {
-                                          label: 'Add join',
-                                          onClick: () => {
-                                              actions.selectSourceTable(table.name)
-                                              actions.toggleJoinTableModal()
-                                          },
-                                      },
-                                  ],
-                              }))
-                            : dataWarehouseTablesBySourceType,
-                    onAdd: () => {
-                        router.actions.push(urls.pipeline(PipelineTab.Sources))
-                    },
-                } as SidebarCategory,
-                {
-                    key: 'data-warehouse-tables',
-                    noun: ['table', 'tables'],
-                    loading: databaseLoading,
-                    items: relevantPosthogTables.map(([table, matches]) => ({
-                        key: table.id,
-                        name: table.name,
-                        url: '',
-                        searchMatch: matches
-                            ? {
-                                  matchingFields: matches.map((match) => match.key),
-                                  nameHighlightRanges: matches.find((match) => match.key === 'name')?.indices,
-                              }
-                            : null,
-                        onClick: () => {
-                            actions.selectSchema(table)
-                        },
+                databaseLoading,
+                folders
+            ) => {
+                // Helper to build nested folder structure
+                const buildFolderTree = (parentId: string | null = null): ListItemAccordion[] => {
+                    return folders
+                        .filter((folder) => folder.parent === parentId)
+                        .map((folder) => {
+                            const folderItems = folder.items
+                                .map((itemId: string) => {
+                                    const savedQuery = dataWarehouseSavedQueries.find((q) => q.id === itemId)
+                                    if (!savedQuery) {
+                                        return null
+                                    }
+                                    return {
+                                        key: savedQuery.id,
+                                        name: savedQuery.name,
+                                        url: '',
+                                        icon: savedQuery.status ? (
+                                            <Tooltip title="Materialized view">
+                                                <IconCalculate />
+                                            </Tooltip>
```

Comment:
I'm having console errors flushed and I think they are related to this one?
<img width="517" alt="image" src="https://github.com/user-attachments/assets/be0eb9b2-897b-477c-8591-00520cb5cacf" />